bin/portal-bridge/src/bridge/e2hs.rs

```diff
@@ -0,0 +1,439 @@
+            let path = entry.path();
+            let path_str = path.to_str()?;
+            if path_str.contains(&format!("mainnet-{epoch_index:05}")) {
+                Some(path_str.to_string())
+            } else {
+                None
+            }
+        })
+        .collect();
+    let file_count = files.len();
+    ensure!(
+        file_count == 1,
+        "Expected 1 file for epoch {epoch_index}, found {file_count}"
+    );
+    let file = files.first().expect("to be able to get first file");
+    Ok(file.clone())
+}
+
+#[derive(Debug)]
+pub struct E2HSBridge {
+    portal_client: HttpClient,
+    metrics: BridgeMetricsReporter,
+    gossip_semaphore: Arc<Semaphore>,
+    header_validator: HeaderValidator,
+    block_range: BlockRange,
+    random_fill: bool,
+}
+
+impl E2HSBridge {
+    pub fn new(
+        portal_client: HttpClient,
+        gossip_limit: usize,
+        block_range: BlockRange,
+        random_fill: bool,
+    ) -> anyhow::Result<Self> {
+        let gossip_semaphore = Arc::new(Semaphore::new(gossip_limit));
+        let mode = format!("{}-{}:{}", block_range.start, block_range.end, random_fill);
+        let metrics = BridgeMetricsReporter::new("e2hs".to_string(), &mode);
+        let header_validator = HeaderValidator::new();
+        Ok(Self {
+            portal_client,
+            metrics,
+            gossip_semaphore,
+            header_validator,
+            block_range,
+            random_fill,
+        })
+    }
+
+    pub async fn launch(&self) {
+        info!("Launching E2HS bridge");
+        let epochs = block_range_to_epochs(self.block_range.start, self.block_range.end);
+        let mut epoch_indexes: Vec<u64> = epochs.keys().cloned().collect();
+        if self.random_fill {
+            epoch_indexes.shuffle(&mut rand::thread_rng());
+        }
+        for index in epoch_indexes {
+            let block_range = epochs.get(&index).expect("to be able to get block range");
+            self.gossip_epoch(index, *block_range).await;
+        }
+    }
+
+    async fn gossip_epoch(&self, epoch_index: u64, block_range: Option<(u64, u64)>) {
+        match block_range {
+            Some(_) => {
+                info!("Gossiping block range: {block_range:?} from epoch {epoch_index}",);
+            }
+            None => {
+                info!("Gossiping entire epoch {epoch_index}");
+            }
+        }
+        let Ok(e2hs_path) = get_e2hs_file(epoch_index) else {
+            panic!("Failed to find e2hs file for epoch {epoch_index}");
+        };
+        let raw_e2hs = std::fs::read(e2hs_path).expect("Failed to read e2hs file");
+        let mut serve_block_tuple_handles = vec![];
+        let block_stream = E2HS::iter_tuples(raw_e2hs).expect("to be able to iter tuples");
+        for block_tuple in block_stream {
+            let block_number = block_tuple
+                .header_with_proof
+                .header_with_proof
+                .header
+                .number;
```

Comment:
We are doing this 3 times with 3 different variable names
- block_number
- number
- header_number

Could we only do it once, then just pass it as a variable and use `block_number` as I think that is the standard terminology.